app/controllers/action/AuthorisedSaUserAction.scala

```diff
@@ -18,51 +18,66 @@ package controllers.action
 
 import com.google.inject.Inject
 import config.AppConfig
-import play.api.Logger
+import journey.Journey
 import play.api.mvc.Results.Redirect
 import play.api.mvc._
 import timetopaytaxpayer.cor.model.SaUtr
 import uk.gov.hmrc.auth.core.ConfidenceLevel
+import uk.gov.hmrc.selfservicetimetopay.jlogger.JourneyLogger
+import req.RequestSupport._
 
 import scala.concurrent.{ExecutionContext, Future}
-import req.RequestSupport._
-import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 
-final class AuthorisedSaUserRequest[A](val request: AuthenticatedRequest[A], val utr: SaUtr)
-  extends WrappedRequest[A](request)
+final class AuthorisedSaUserRequest[A](val request: AuthenticatedRequestWithJourney[A], val utr: SaUtr)
+  extends WrappedRequest[A](request) {
+
+  val journey: Journey = request.journey
+}
 
 class AuthorisedSaUserAction @Inject() (
     appConfig: AppConfig,
     cc:        MessagesControllerComponents)
-  extends ActionRefiner[AuthenticatedRequest, AuthorisedSaUserRequest] {
-
-  override protected def refine[A](request: AuthenticatedRequest[A]): Future[Either[Result, AuthorisedSaUserRequest[A]]] = {
+  extends ActionRefiner[AuthenticatedRequestWithJourney, AuthorisedSaUserRequest] {
 
+  override protected def refine[A](request: AuthenticatedRequestWithJourney[A]): Future[Either[Result, AuthorisedSaUserRequest[A]]] = {
+    implicit val r: Request[_] = request
+    implicit val ec: ExecutionContext = executionContext
     val hasActiveSaEnrolment: Boolean = request.hasActiveSaEnrolment
     val maybeUtr: Option[SaUtr] = request.maybeUtr
     val isConfident: Boolean = request.confidenceLevel >= ConfidenceLevel.L200
+    import req.RequestSupport._
+
+    //this is being updated by notification from add-taxes-frontend
+    //at some point if user was sent to add-taxes-frontend for enrol-for-sa journey
+    val enrolledForSa = request.journey.userEnrolledForSa.getOrElse(false)
```

Comment:
trivial: name of this val is slightly ambiguous and could be misread/misinterpreted as `isEnrolledForSa` which has a subtlely different meaning